Translating/ModelSBML.py

```diff
@@ -0,0 +1,315 @@
+        :param: items -> list of RHS complexes in reaction"""
+        for itm in items:
+            product = reaction.createProduct()
+            product.setSpecies(itm[0].to_SBML_species_code())
+            product.setConstant(False)
+            product.setStoichiometry(itm[1])
+
+    def create_kinetic_law_and_modifiers(self, rules, reaction_objects):
+        """creates kinetic law for all reactions
+           and at the same time it creates modifiers
+           for those Complexes that were not used either
+           as reactant or product.
+          :param: rules -> BCSL rules
+          :param: reaction_objects -> list of all libsbml reaction objects inorder
+           to work with given reaction in this librialry
+
+         """
+        for i, r in enumerate(rules):
+            reac = r.to_reaction()
+            rate = reac.rate
+            res = rate.to_mathML()
+            agents, params = rate.get_params_and_agents()
+            actors = []
+            for agent in agents:
+                actors.append(agent.to_SBML_species_code())
+            law = reaction_objects[i].createKineticLaw()
+            num_of_products = reaction_objects[i].getListOfProducts().getListOfAllElements().getSize()
+            num_of_reactants = reaction_objects[i].getListOfReactants().getListOfAllElements().getSize()
+
+            law.setMath(libsbml.parseFormula(res))
+            for n in range(num_of_products):
+                product = reaction_objects[i].getListOfProducts().getListOfAllElements().get(n).getSpecies()
+                if product in actors:
+                    actors.pop(actors.index(product))
+            for u in range(num_of_reactants):
+                reactant = reaction_objects[i].getListOfReactants().getListOfAllElements().get(u).getSpecies()
+                if reactant in actors:
+                    actors.pop(actors.index(reactant))
+            for remaining_actor in actors:
+                modifier = reaction_objects[i].createModifier()
+                modifier.setSpecies(remaining_actor)
+
+    def create_reaction_for_isomorphisms(self, unique_complexes):
+        """Function creates all reactions between isomorphism in order to
+        ensure that given Complexes are interchangeable but specific rule
+        applyies only to that exact isomorphism
+
+        :param: unique_complexes
+        """
+        for compl in unique_complexes:
+            if len(unique_complexes[compl]) > 1:
+                for i in range(len(unique_complexes[compl])):
+                    for j in range(len(unique_complexes[compl])):
+                        if i > j:
+                            reaction = self.model.createReaction()
+                            reaction.setId("rc_" + str(unique_complexes[compl][i][1]) + "_to_" + str(unique_complexes[compl][j][1]))
+                            reaction.setName(str(unique_complexes[compl][i][0]) + "_to_" + str(unique_complexes[compl][j][0]))
```

Comment:
Yes, it is the same when checking by eq operator. If this was the only usage it would be useless to store it like that.
Dictionary of `unique_complexes` is used in `create_all_species_compartments_and_complex_species_types` function.
in this function just distingusihing by eq operator is not sufficient. We need to create species for every isomorphism in order to be able to change between their form in model. Storing complex type is good for that reason so SBML species can be created from it in respective function.

Reply:
Alright, I probably didn't notice that it is being used in `create_all_species_compartments_and_complex_species_types` function, where you actually need all present isomorphic forms of the complex.